src/lazy/binary/raw/reader.rs

```diff
@@ -188,12 +200,41 @@ impl<'data> DataSource<'data> {
         self.bytes_to_skip = lazy_value.encoded_value.total_length();
         Ok(Some(lazy_value))
     }
+
+    /// Runs the provided parsing function on this DataSource's buffer.
+    /// If it succeeds, marks the `DataSource` as ready to advance by the 'n' bytes
+    /// that were consumed.
+    /// If it does not succeed, the `DataSource` remains unchanged.
+    pub(crate) fn try_parse_next_field<
```

Comment:
🗺️ Now that struct fields are their own type (and not just a different "view" of a value), they need their own parsing method on the reader.